apisix/core/pubsub.lua

```diff
@@ -0,0 +1,131 @@
+--
+-- Licensed to the Apache Software Foundation (ASF) under one or more
+-- contributor license agreements.  See the NOTICE file distributed with
+-- this work for additional information regarding copyright ownership.
+-- The ASF licenses this file to You under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with
+-- the License.  You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+
+local core         = require("apisix.core")
+local ws_server    = require("resty.websocket.server")
+local protoc       = require("protoc")
+local pb           = require("pb")
+local setmetatable = setmetatable
+local pcall        = pcall
+local pairs        = pairs
+
+protoc.reload()
+pb.option("int64_as_string")
```

Comment:
Will this call pollute the global option of pb?

Reply:
@spacewander 

I think it's possible that this `pb` module is a C module imported from `pb.so`, which is perhaps loaded only once; I can't be sure. 🤔

BTW, user-defined option lists are allowed in `grpc-transcode`, so if this is possible here, `grpc-transcode` will also have problems (multiple different grpc routes allow different options to be defined)

Reply:
Maybe we can solve with the pb_State like how we store the proto in grpc-transcode?

Reply:
**I need to research this further and will update it later if needed, you can review the other code first.**

Reply:
I help you to do it today, and put the result as: https://github.com/apache/apisix/issues/7025

Reply:
fixed #7028

Reply:
resolved as `grpc-transcode` in #7028